lib/services/authentication-service.ts

```diff
@@ -23,19 +30,20 @@ export class AuthenticationService implements IAuthenticationService {
 		this.$logger.info("Launching login page in browser.");
 
 		let loginUrl: string;
-		let localhostServer = this.$httpServer.createServer({
+		this.localhostServer = this.$httpServer.createServer({
 			routes: {
 				"/": async (request: ServerRequest, response: ServerResponse) => {
 					this.$logger.debug("Login complete: " + request.url);
 					const parsedUrl = parse(request.url, true);
 					const loginResponse = parsedUrl.query.response;
 					if (loginResponse) {
 						await this.serveLoginFile("end.html")(request, response);
-						localhostServer.close();
+						this.localhostServer.close();
```

Comment:
Do you think it appropriate to set `this.localhostServer = null;` here?
If so you can extract this functionality and use it in `cancelLogin` down below too